pip/req/req_file.py

```diff
@@ -51,7 +51,8 @@
 # options to be passed to requirements
 SUPPORTED_OPTIONS_REQ = [
     cmdoptions.install_options,
-    cmdoptions.global_options
+    cmdoptions.global_options,
+    cmdoptions.hash,
 ]
```

Comment:
understood, but in general, we try to have options work in the requirements file for consistency.

supporting it in the requirements file allows "writers" of requirements file to declare an intent for the file prior to knowing any specific hash.  as it is, that intention only exists by applying the option in the cli, external to the file.